Amplify/Categories/Auth/Models/AuthCodeDeliveryDetails.swift

```diff
@@ -0,0 +1,34 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+public struct AuthCodeDeliveryDetails {
+
+    public let destination: String
+
+    public let deliveryMedium: DeliveryMedium
+
+    public let attributeName: String
+
+    public init(destination: String,
+                 deliveryMedium: DeliveryMedium,
+                 attributeName: String) {
+        self.destination = destination
+        self.deliveryMedium = deliveryMedium
+        self.attributeName = attributeName
+    }
+
+}
+
+//TODO: Should we move this under AuthCodeDeliveryDetails extension?
+public enum DeliveryMedium {
+    case sms
+    case email
+    case phoneNumber
+    case unknown(String)
```

Comment:
i'm thinking this should be `.custom(String)` as well since this developer instantiated intent of "adding a custom deliivery medium", isn't really like the other case where it is an unknown error case.

Reply:
Added, might change after review